c/datasetjson.c

```diff
@@ -1451,6 +1449,24 @@ void updateDataset(HttpResponse* response, char* absolutePath, int jsonMode) {
     respondWithError(response,HTTP_STATUS_INTERNAL_SERVER_ERROR,"Could not translate character set to EBCDIC");    
   }
   safeFree(convertedBody,conversionBufferLength);
+
+  if(rc != -1) {
```

Comment:
if changing the code to have more/different rcs, its often the case we just do `if (rc != 0)`